markups/commonmark.py

```diff
@@ -0,0 +1,56 @@
+# vim: ts=8:sts=8:sw=8:noexpandtab
+
+# This file is part of python-markups module
+# License: BSD
+# Copyright: (C) Shengjing Zhu, 2016
+
+import markups.common as common
+from markups.abstract import AbstractMarkup, ConvertedMarkup
+
+
+class CommonMarkMarkup(AbstractMarkup):
+	"""Markup class for CommonMark language.
+	Inherits :class:`~markups.abstract.AbstractMarkup`.
+	"""
+	name = 'CommonMark'
+	attributes = {
+		common.LANGUAGE_HOME_PAGE: 'http://commonmark.org/',
+		common.MODULE_HOME_PAGE: 'https://github.com/rtfd/CommonMark-py/',
+		common.SYNTAX_DOCUMENTATION: 'http://spec.commonmark.org/'
+	}
+
+	# Note: the same as MarkdownMarkup
+	# file_extensions = ('.md', '.mkd', '.mkdn', '.mdwn', '.mdown', '.markdown')
+	# default_extension = '.md'
+
+	@staticmethod
+	def available():
+		try:
+			import CommonMark
+		except ImportError:
+			return False
+		return True
+
+	def __init__(self, filename=None, options={}):
+		self.enable_highlight = options.get('highlight', False)
+		super().__init__(filename)
+		import CommonMark
+		from markups.cmark_render import CommonMarkRender
+		self.parse = CommonMark.Parser().parse
+		if self.enable_highlight:
+			self.render = CommonMarkRender(options).render
+		else:
+			self.render = CommonMark.HtmlRenderer(options).render
+
+	def convert(self, text):
+		ast = self.parse(text)
+		html = self.render(ast)
+		if self.enable_highlight:
+			stylesheet = common.get_pygments_stylesheet('.highlight')
+		else:
+			stylesheet = ''
+		return CovertedCommonMark(html, stylesheet=stylesheet)
+
+
+class CovertedCommonMark(ConvertedMarkup):
+	pass
```

Comment:
Just a workaround to avoid Retext's warning.

```
Traceback (most recent call last):
  File "/home/zsj/Workspaces/py/retext/ReText/tab.py", line 228, in updatePreviewBox
    self.converted = self.converterProcess.get_result()
  File "/home/zsj/Workspaces/py/retext/ReText/converterprocess.py", line 172, in get_result
    raise ConversionError(result)
ReText.converterprocess.ConversionError: The background markup conversion process received this exception:
    Traceback (most recent call last):
      File "/home/zsj/Workspaces/py/retext/ReText/converterprocess.py", line 77, in _converter_process_func
        converted = current_markup.convert(job['text'])
      File "/home/zsj/Workspaces/py/pymarkups/markups/commonmark.py", line 52, in convert
        return CovertedCommonMark(html, stylesheet=stylesheet)
    NameError: name 'CovertedCommonMark' is not defined
```

Remove it will cause a lot of warnings in Retext.